### DISCUSSION TITLE
Filter Datetime

### DISCUSSION THREAD

**Discussion by gladtoseeuhappy** [carltongibson/django-filter#1387] [Q&A]
i have ***json datetime*** like this
```
{
    "datetime": "2020-09-11 11:09:52",
}
```
I tried to ***filter*** it something like this 

***localhost:8000/api/v1/history?datetime=2020-09-11%2011:09:52.***

<b>`Here what i tried`</b>
```
class AuditlogFilter(filters.FilterSet):
    datetime = filters.DateTimeFilter(method="datetime_filter")

    class Meta:
        model = CRUDEvent
        fields = ['datetime']

    def datetime_filter(self, queryset, name, value):
        return queryset.filter(Q(datetime__icontains=value))

```

However i got ***empty value***. Is there a way  i can **archieve** it??? 

Thanks in advance...

**Comment by AviKKi** [comment]
is datetime a CharField? 

`__icontains` only works for Char and Text Field, use `__contains` instead for a datetimefield.

Also searching by exact time till the second may not give results try below instead

`localhost:8000/api/v1/history?datetime=2020-09-11`

also try to use builtin methods like `DateFilter`, `TimeFilter`, `IsoDateTimeFilter`

**Comment by king-11** [comment]
@N0SYS73M1SS4F3 you are using an ISO Date Time value so maybe your filter should be IsoDateTimeFilter

**Comment by rickerp** [comment]
I have the same problem, with a similar structure. The datetime is a `models.DateTimeField(blank=False)` and my `method` function has a print in it, but never prints anything which leads me to think that never calls it. My guess is that datetime is being associated with the model field and creates a filter Field automatically which ignore the one manually declared that has the method associated (just a guess)

**Comment by panosangelopoulos** [comment]
@rickerp Did you find a way to solve it? If this is still an open issue could you please share some code examples, it will be easier to understand the root of the problem.
